pkg/controller/core/resourceflavor_controller.go

```diff
@@ -79,11 +79,11 @@ func (r *ResourceFlavorReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		}
 	} else {
 		if controllerutil.ContainsFinalizer(&flavor, kueue.ResourceInUseFinalizerName) {
-			if cqName, ok := r.cache.FlavorInUse(flavor.Name); ok {
-				log.V(3).Info("resourceFlavor is still in use", "ClusterQueue", cqName)
+			if cqs := r.cache.ClusterQueuesUsingFlavor(flavor.Name); cqs != nil {
```

Comment:
We generally avoid having a distinction between empty slice and nil.

```suggestion
			if cqs := r.cache.ClusterQueuesUsingFlavor(flavor.Name); len(cqs) != 0 {
```

Reply:
Makes sense. I have updated it.